src/main.rs

```diff
@@ -741,49 +728,49 @@ impl ManParser for Type3 {
         manpage.contains(".SH DESCRIPTION")
     }
 
-    fn parse_man_page(&mut self, manpage: &str) -> Option<String> {
-        unimplemented!();
+    fn parse_man_page(&self, manpage: &str, cmdname: &str) -> Option<String> {
+        let options_section_re = regex!(r"\.SH DESCRIPTION((?s:.)*?)(\.SH|\\Z)");
+        let options_section_matched = options_section_re.find(manpage);
+        let mut options_section = options_section_matched.unwrap().as_str();
+
+        let options_parts_re = regex!(r"\.TP((?s:.)*?)\.TP");
+        let mut options_matched = options_parts_re.captures(options_section);
+        // add_diagnostic(format!("Command is {}", cmdname));
+
+        if options_matched.is_none() {
+            // add_diagnostic("Unable to find options section");
+            return None;
+        }
+
+        let mut completions = Completions::new(cmdname);
+        while let Some(mat) = options_matched {
+            let mut data = mat.get(1).unwrap().as_str();
+
+            let data = remove_groff_formatting(data);
+            let data = data.trim();
+            let data: Vec<&str> = data.splitn(2, '\n').collect();
+
+            if data.len() < 2 {
+                // add_diagnostic("Unable to split option from description");
+                return None;
+            }
+            let option_name = data[0].trim();
+            if option_name.contains('-') {
+                let option_name = unquote_double_quotes(option_name);
+                let option_name = unquote_single_quotes(option_name);
+                let option_desc = data[1].trim().replace("\n", " ");
+                completions.add(&option_name, &option_desc);
```

Comment:
I _think_ this would work but haven't tested it locally. Wanted to mimic the `next_tuple` pattern you had used elsewhere.
```suggestion
            let (option_name, option_desc) = match data.splitn(2, '\n').next_tuple() {
                Some(tuple) => tuple,
                None => {
                    // add_diagnostic("Unable to split option from description");
                    return None;
                }
            };
            let option_name = option_name.trim();
            if option_name.contains('-') {
                let option_name = unquote_double_quotes(option_name);
                let option_name = unquote_single_quotes(option_name);
                let option_desc = option_desc.trim().replace("\n", " ");
                completions.add(&option_name, &option_desc);
```

Reply:
Oh, I did try that. The reason why I did not used that method is because it is longer, 2-3 lines last time I tried.